### ISSUE TITLE
Create UI to allow any component to be added to a node

### ISSUE THREAD

**Issue by collinsfoobar** [besasm/EMGAATS#111]
Storage, infiltration can be added with a button. Weirs, orifices, pumps have spatial component. Easiest way is spatially for user.

User feedback needed for spatial vs tabular...present consequences of doing either way to user, and have them decide.

Includes selecting nodes/links graphically, which will be used in other property dialogs (e.g., weirs)

__Based on user feed back:__
Select link from pull down list of valid links
Select node graphically

**Comment by ahedhi** [comment]
For named components, how do we auto-name them when created in the UI?

**Comment by arnelmandilag** [comment]
How about maintaining a system table that holds the next available ID (a long) for all the different types of named components? We don't want to re-use IDs, and this field isn't an identity field. The name will be just the number.

**Comment by ahedhi** [comment]
Actually, I think for the ones I'm thinking about (Storage and Infiltrator), I think we agreed to remove these. Also, for weirs, orifices and pumps, I believe there's an existing convention related to which link it is associated with?

**Comment by arnelmandilag** [comment]
Yeah, I remember you guys saying last week that names would be eliminated for Storage and Infiltrator, as there's at most one per node. For the others, you are somewhat correct (see entry at https://github.com/besasm/EMGAATS/wiki/FrameworkDevelopmentBlog#virtual-nodes). This entry was for what to do at deployment time and virtual node creation. We'll want to come up with some kind of mechanism for coming up with names on weirs, orifices, and pumps. In the past, we used a similar mechanism to what I proposed (we only had one table combining the three types, so we just used the max+1 special link ID number for the name of the object).
